src/main/java/org/spongepowered/api/command/Command.java

```diff
@@ -95,7 +96,7 @@ static Builder builder() {
      * @return The result of a command being processed
      * @throws CommandException Thrown on a command error
      */
-    CommandResult process(CommandCause cause, String arguments) throws CommandException;
+    CommandResult process(CommandCause cause, ArgumentReader.Mutable arguments) throws CommandException;
```

Comment:
A comment rather than a request:

I'm very "eh" on this. I'm not really in favour of forcing the use of the reader on commands, but on the other hand, I understand the reasons why you'd want to provide it. I'd prefer being to create a reader from a string for use elsewhere too (if someone wanted to parse a chat message or something, for example), and keep this as a string so developers can decide, but it's a design/preference sort of thing here. 

I don't care enough to argue about it though because this may very well be the preferred direction anyway. I just see "Raw" as being as low level as possible, which this effectively prevents, but as you can get the string out of the reader, it's not _really_ an issue.

Reply:
hm, this was something I was unsure about as well at first. Why I eventually settled on exposing the argument reader as a method parameter was that we already allow Raw commands to generate a command tree, and an argument reader is an easy way to parse input in a way that is compatible with the types usable for command trees.

Because the raw string is available in `arguments.getRemaining()` this doesn't lock raw commands out of any ways of working, but does give them a pointer towards the right direction to go when seeking low-level access.